src/Geno/C++/GUI/Widgets/TitleBar.cpp

```diff
@@ -207,6 +211,66 @@ void TitleBar::Draw( void )
 				ButtonRect.Min.x -= ButtonSize;
 				ButtonRect.Max.x -= ButtonSize;
 			}
+
+		#if defined ( __linux__ )
+			// X11 Window Move
+			{
+				GLFWwindow* pWindow          = MainWindow::Instance().GetWindow();
+				Display* pX11Display      = glfwGetX11Display();
+				Window      X11Window        = glfwGetX11Window( pWindow );
+				ImVec2      CursorPos        = ImGui::GetMousePos();
+				int         WindowSize[ 2 ]  ={ 0, 0 };
+				int         Border           = 5;
+
+				glfwGetWindowSize( pWindow, &WindowSize[ 0 ], &WindowSize[ 1 ] );
+
+				int rx, ry;
+
+				XWindowAttributes win_attributes;
+
+				Window junkwin;
+
+				XGetWindowAttributes( pX11Display, X11Window, &win_attributes );
+
+				// Get Top left
+				XTranslateCoordinates( pX11Display, X11Window, win_attributes.root, -win_attributes.border_width, -win_attributes.border_width, &rx, &ry, &junkwin );
+				Rect WindowRect;
+				WindowRect.x1 = rx;
+				WindowRect.y1 = ry;
+				WindowRect.x2 = WindowSize[ 0 ];
+				WindowRect.y2 = WindowSize[ 1 ];
+
+				if( ImGui::IsMousePosValid( &CursorPos ) )
+				{
+					if( CursorPos.y < ( WindowRect.Top() + Border ) )
+					{
+						if( CursorPos.x < ( WindowRect.Left() + Border ) ) { ImGui::SetMouseCursor( ImGuiMouseCursor_ResizeNWSE ); glfwX11ResizeWindow( pWindow, 5 ); }
+						else if( CursorPos.x >= ( WindowRect.Right() - Border ) ) { ImGui::SetMouseCursor( ImGuiMouseCursor_ResizeNESW ); glfwX11ResizeWindow( pWindow, 6 ); }
+					}
+					else if( CursorPos.y >= ( WindowRect.Bottom() - Border ) )
+					{
+						if( CursorPos.x < ( WindowRect.Left() + Border ) ) { ImGui::SetMouseCursor( ImGuiMouseCursor_ResizeNESW ); glfwX11ResizeWindow( pWindow, 7 ); }
+						else if( CursorPos.x >= ( WindowRect.Right() - Border ) ) { ImGui::SetMouseCursor( ImGuiMouseCursor_ResizeNWSE ); glfwX11ResizeWindow( pWindow, 8 ); }
+					}
+					else if( CursorPos.x < ( WindowRect.Left() + Border ) )
+					{
+						ImGui::SetMouseCursor( ImGuiMouseCursor_ResizeEW );
+						glfwX11ResizeWindow( pWindow, 1 );
+					}
+					else if( CursorPos.x >= ( WindowRect.Right() - Border ) )
+					{
+						ImGui::SetMouseCursor( ImGuiMouseCursor_ResizeEW );
+						glfwX11ResizeWindow( pWindow, 3 );
+					}
+					else
+					{
+						// Drag the menu bar to move the window
+						if( !ImGui::IsAnyItemHovered() && ( CursorPos.y < ( WindowRect.Top() + m_Height ) ) && ImGui::IsMouseClicked( ImGuiMouseButton_Left ) )
+							glfwDragWindow( pWindow );
```

Comment:
I don't think we should pretend to be GLFW and prefix our custom functions this way. It leads me to believe that this was built into GLFW and that we didn't put effort into implementing it ourselves.